framework/libs/ktest/src/runner.rs

```diff
@@ -110,35 +120,41 @@ where
                     continue;
                 }
             }
-            print!(
-                "test {}::{} ...",
-                test.info().module_path,
-                test.info().fn_name
-            );
+
             debug_assert_eq!(test.info().package, crate_name);
-            match test.run(catch_unwind) {
-                Ok(()) => {
-                    print!(" {}\n", "ok".green());
-                    passed += 1;
-                }
-                Err(e) => {
-                    print!(" {}\n", "FAILED".red());
-                    failed_tests.push((test.clone(), e.clone()));
-                }
-            }
+
+            spawn_ktest(deps, test);
```

Comment:
Spawning too many tasks is not an ideal configuration. A more sensible configuration is to have as many tasks as the number of CPUs. `cargo test` even allow the user to specify the concurrency at run time (e.g., `cargo test -- --test-threads=2`). I think the ktest framework should take such an argument.

Reply:
Currently (spawning == creating and running immediately) and no preemption is allowed. So it would be main -> test1 -> main -> test2 -> main -> test3 ... currently in single CPU.

It is not a big deal of problem currently I guess. But supporting test workers is indeed better.